packages/salesforcedx-vscode-core/package.json

```diff
@@ -156,6 +157,10 @@
         }
       ],
       "editor/context": [
+        {
+          "command": "sfdx.force.function.start",
+          "when": "sfdx:functions_enabled && sfdx:project_opened && resource =~ /.*/functions/[^/]+(/[^/]+\\.(ts|js|json|toml))?$/"
```

Comment:
I think this when clause needs to be tweaked a bit more. It currently shows the command if you right click on the top level `functions` folder, ideally it should only be shown on the child folders that represent an individual function.

We should try and limit the command to be shown when right clicking only on those file that do represent the actual function code (index, .toml, etc). It is currently shown when right-clicking on files like package-lock.json or tsconfig.json

Reply:
> It currently shows the command if you right click on the top level functions folder, ideally it should only be shown on the child folders that represent an individual function.

It doesn't show up in the top level folder
<img width="497" alt="Screen Shot 2020-09-15 at 6 03 21 PM" src="https://user-images.githubusercontent.com/679275/93280202-e7ca7800-f77d-11ea-8d3a-d85e2e5946f0.png">

>We should try and limit the command to be shown when right clicking only on those file that do represent the actual function code (index, .toml, etc). It is currently shown when right-clicking on files like package-lock.json or tsconfig.json

I added some limits in https://github.com/forcedotcom/salesforcedx-vscode/pull/2498/commits/49e287e2380861c76b66118964ce5afd0cd8ee46 I think this is less concerning than similar limits in CodeLens though, since it actually makes sense to start function from _any_ file inside the function folder, and currently the command is available only in right click menu.